jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitRunner.kt

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.CapturingProcessHandler
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.settings.SamSettings
+import software.aws.toolkits.resources.message
+
+class SamInitRunner {
+    private val samCliExecutable = SamSettings.getInstance().executablePath
+    private val commandLine = GeneralCommandLine(samCliExecutable)
+            .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
+            .withParameters("init")
+            .withParameters("--no-input")
+    private lateinit var outputDir: VirtualFile
+    private val parameters = mutableMapOf<String, String>()
+
+    fun applyLocation(location: String) = applyParameter("--location", location)
+
+    fun applyName(name: String) = applyParameter("--name", name)
+
+    fun applyOutputDir(location: VirtualFile) = apply {
+        outputDir = location
```

Comment:
Is there a reason these aren't applied / sourced in a constructor? Why do we have to have these mutation methods?

Reply:
awesome!